### PR TITLE
Add UI, Luhn Test, Responsive and dark mode

### PR DESCRIPTION
**Pull request by NagaSatyaGeeth**
Made it pretty. Added Luhn Test as well. Made it mobile friendly too and dark mode as well. 

You can check out the live website here https://cc-billpay-upi-id.vercel.app/

I dont want another website with the same functionality running so i will take down the link i have given here once a decision about this PR has been made. I only added this link so that it is easy for you to check it out without downloading it and going through the hassle.

image

image

Had to use React so let me know if you dont know how to host react apps on github pages.

### CODE REVIEW

**Comment by NagaSatyaGeeth** [comment]
dummy cc number incase you want to test Luhn algo i have implemented: 4417 1234 5678 9113
I grabbed it from here https://www.groundlabs.com/blog/anatomy-of-a-credit-card/

**Comment by NagaSatyaGeeth** [comment]
Things to improve upon in the code i have submitted: 

Break it down into more components. 

I would totally understand if you would like to keep the code only in pure html css and js however I would also suggest you to breakdown your code into different files for better readability.

**Comment by mkagenius** [comment]
This looks nice. I can merge. But give me some time to review this, there are a lot of changes to go through. The QR code bit is not rebased in this one?

I will also need one more person to review this. Let me see.

**Comment by NagaSatyaGeeth** [comment]
> This looks nice. I can merge. But give me some time to review this, there are a lot of changes to go through. The QR code bit is not rebased in this one?
> 
> I will also need one more person to review this. Let me see. 
> 


No I didn't add the QR code feature because I've realised it was added only today morning after I've pushed the code. I can't confirm if I will be able to add the QR code feature any soon as I've the other projects that require my attention right now. 

I can refactor the code into different files if needed for now but that too will take like 3 or 4 days.

File: cc-billpay-upi-id/index.html [hunk 1]
```diff
@@ -0,0 +1,13 @@
+<!doctype html>
+<html lang="en">
+  <head>
+    <meta charset="UTF-8" />
+    <link rel="icon" type="image/svg+xml" href="/vite.svg" />
+    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
+    <title>Vite + React + TS</title>
```
Comment (mkagenius) [inline cc-billpay-upi-id/index.html:7]: Need to retain the original title. If not in commit then maybe during merge or rebase

File: cc-billpay-upi-id/public/vite.svg
Comment (mkagenius) [inline cc-billpay-upi-id/public/vite.svg:1]: if this is not used, we can remove this.

File: cc-billpay-upi-id/src/App.tsx [hunk 2]
```diff
@@ -0,0 +1,285 @@
...
+			document.body.removeChild(tempTextArea);
+		}
+	};
+
+	return (
+		<main className="min-h-screen bg-gradient-to-br from-purple-100 to-indigo-200 dark:bg-gradient-to-br dark:from-purple-900 dark:to-indigo-900 flex items-center justify-center py-10">
+			<div className="container relative sm:mt-0 max-w-4xl bg-white dark:bg-gray-800 dark:text-gray-100 p-8 rounded-lg overflow-auto">
+				<div className="flex justify-between items-center mb-6">
+					<div className="heading text-2xl font-bold text-purple-800 dark:text-purple-300">
+						Pay Your Credit Card Bills via UPI
```
Comment (mkagenius) [inline cc-billpay-upi-id/src/App.tsx:153]: note: subtitle missing. add it after the merge.

File: cc-billpay-upi-id/src/App.tsx [hunk 3]
```diff
@@ -0,0 +1,285 @@
...
+													{/* Buttons - Flex container with wrap */}
+													<div className="flex flex-wrap gap-2">
+														<button
+															className="rounded-md text-black border-2 border-gray-400 bg-white dark:text-gray-300 dark:bg-gray-600 dark:border-gray-600 dark:hover:bg-gray-600 flex items-center p-1"
+															onClick={() => copyToClipboard(upiId)}
+															disabled={disabled}
+														>
+															<Copy className="w-4 h-4 mr-1" />{" "}
+															{copiedUPIId === upiId ? "Copied" : "Copy"}
+														</button>
```
Comment (mkagenius) [inline cc-billpay-upi-id/src/App.tsx:252]: QR code feature is missing. Maybe add qr code after the merge but that will be a major regression - so i need to not push the code to prod until we have the qr also in the code. lets see.

**Review by mkagenius — COMMENTED** [review]
Don't see any security issue. I will try to merge locally and add the QR code and add new subtitle/title before pushing it live though.

**Comment by mkagenius** [comment]
created a new branch to work on this merge - https://github.com/RedeemApp/cc-billpay-upi-id/tree/react-app-branch

Added qr code.

Todo:
1. Autocomplete not working in this, need to enable that.

2. Code is complex, make it more readable, it will be easier to build trust.

**Pull request merged**